### ISSUE TITLE
Flat for tensors other than 4 dimensions

### ISSUE THREAD

**Issue by soumyac1999** [flexflow/FlexFlow#592]
Currently flat is hard-coded for 4 dimensional tensors. It will be good to have more general support.
https://github.com/flexflow/FlexFlow/blob/e2796ff7a874fbb9cfcffe6ba0f3af13239a6e3f/src/ops/flat.cc#L41

**Comment by jiazhihao** [comment]
`Flat` is a special case of `Reshape`, which supports general changes to a tensor's dimensions

**Comment by jiazhihao** [comment]
The semantic of `BatchNorm` is equivalent to PyTorch's BatchNorm2d: https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm2d.html. Are you thinking of 5d or 6d batchnorm?

**Comment by soumyac1999** [comment]
For Flat, should I then change the Keras API for Flatten to use Reshape instead? So that we are consistent with actual Keras. Or would that affect performance?
Maybe I could make Flatten use Flat for 4 dimensional tensors and Reshape for others.

**Comment by soumyac1999** [comment]
> The semantic of `BatchNorm` is equivalent to PyTorch's BatchNorm2d: https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm2d.html. Are you thinking of 5d or 6d batchnorm?

I was thinking of batchnorm for 2d dimensional tensors (batch_size, hidden size).

**Comment by jiazhihao** [comment]
Reshape should achieve the same performance as Flat.

**Comment by jiazhihao** [comment]
> I was thinking of batchnorm for 2d dimensional tensors (batch_size, hidden size).

Is there a PyTorch-equivalent operator for what you want to do?
